Applications/FileIO/SWMM/SWMMInterface.h

```diff
@@ -0,0 +1,245 @@
+ /**
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ */
+
+#ifndef SWMMINTERFACE_H_
+#define SWMMINTERFACE_H_
+
+#include <iostream>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <logog/include/logog.hpp>
+
+#include "ThirdParty/SWMMInterface/swmm5_iface.h"
+
+#include "GeoLib/Station.h"
+
+namespace GeoLib {
+    class GEOObjects;
+    class Point;
+    class Polyline;
+}
+
+namespace MeshLib {
+    class Mesh;
+    class Node;
+    class Element;
+}
+
+/// SWMM object types
+enum class SwmmObject
+{
+    SUBCATCHMENT = 0,
+    NODE = 1,
+    LINK = 2,
+    SYSTEM = 3
+};
+
+/**
+ * Interface for reading files used within the  * Storm Water Management Model (SWMM) and
+ * converting the data therein into corresponding OGS data structures.
+ * SWMM distinguishes four different object types (defined in the SwmmObject enum):
+ *     * Subcatchments
+ *     * Nodes (or junctions)
+ *     * Links (or conduits)
+ *     * System
+ * Note that each object in a SWMM input file has a name.
+ * Each of the four object types has a (different) number of parameters. There can be an arbitrary
+ * number of nodes, links or subcatchments but only ever one system.
+ * The interface can convert the SWMM input data into a geometry or a (line-)mesh.
+ * For meshes, also output data can be added to nodes and elements.
+ * The interface also provides routines for returning data as vectors as well as convenience
+ * methods for outputting data into CSV files. CSV files will either contain all parameters for one
+ * object at all time steps or all parameters for all objects of a given type at one timestep.
+ */
+class SwmmInterface
+{
+public:
+    /// Basic method to create the interface (containing a mesh) from a SWMM file
+    /// Other non-static methods rely on this being called in the beginning.
+    static SwmmInterface* create(std::string const& file_name);
+
+    /// If a mesh has already been created, this methods allows to add node- or link-arrays as property
+    /// to that mesh. The data vectors can be created using the getArrayAtTimeStep() method.
+    static bool addResultsToMesh(MeshLib::Mesh &mesh, SwmmObject const type,
+        std::string const& vec_name, std::vector<double> const& data);
+
+    /// Returns the mesh generated from SWMM file content.
+    MeshLib::Mesh* getMesh() const { return _mesh.get(); }
```

Comment:
I'd say it differently: Return a reference if the mesh must not be a nullptr, otherwise the pointer is OK (for me the ownership obviously stays with the SwmmInterface)